crates/fuel-core/src/database.rs

```diff
@@ -167,10 +178,17 @@ unsafe impl Send for Database {}
 unsafe impl Sync for Database {}
 
 impl Database {
+    pub fn new(data_source: DataSource) -> Self {
+        Self {
+            data: data_source,
+            _drop: Default::default(),
+        }
+    }
+
     #[cfg(feature = "rocksdb")]
-    pub fn open(path: &Path) -> DatabaseResult<Self> {
+    pub fn open(path: &Path, capacity: Option<usize>) -> DatabaseResult<Self> {
```

Comment:
```suggestion
    pub fn open(path: &Path, capacity: impl Into<Option<usize>>) -> DatabaseResult<Self> {
```
Then the api user can pass in `open(path, 1000)` or `open(path, None)`